README.md

```diff
@@ -147,6 +147,68 @@ These types are unsupported as function parameter types or return types for func
 * [`void`](https://www.typescriptlang.org/docs/handbook/2/functions.html#void), [`object`](https://www.typescriptlang.org/docs/handbook/2/functions.html#object), [`unknown`](https://www.typescriptlang.org/docs/handbook/2/functions.html#unknown), [`never`](https://www.typescriptlang.org/docs/handbook/2/functions.html#never), [`any`](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#any) types - to accept and return arbitrary JSON, use `sdk.JSONValue` instead
 * `null` and `undefined` - unless used in a union with a single other type
 
+### Error handling
+By default, unhandled errors thrown from functions are caught by the Lambda SDK host, and an `InternalServerError` is returned to Hasura. The details of the uncaught error (the message and stack trace) is captured and will be logged in the OpenTelemetry trace associated with the GraphQL request. However, the GraphQL API caller will receive a generic "internal error" response to their query. This is to ensure internal error details are not leaked to GraphQL API clients.
+
+If you want to return specific error details to the GraphQL API client, you can deliberately throw one of the below error classes (these error types correspond with the error status codes in the [NDC Specification](https://hasura.github.io/ndc-spec/specification/error-handling.html)):
+
+| Error Class              | Used When |
+|--------------------------|-----------|
+| sdk.Forbidden            | A permission check failed - for example, a mutation might fail because a check constraint was not met. |
+| sdk.Conflict             | A conflicting state would be created for the data source - for example, a mutation might fail because a foreign key constraint was not met. |
+| sdk.UnprocessableContent | There was something semantically incorrect in the request. For example, an invalid value for a function argument was received. |
+
+```typescript
+import * as sdk from "@hasura/ndc-lambda-sdk"
+
+/** @readonly */
+export function divide(x: number, y: number): number {
+  if (y === 0) {
+    throw new sdk.UnprocessableContent("Cannot divide by zero", { myErrorMetadata: "stuff", x, y })
```

Comment:
Can you include any value in the details, or just objects?

Reply:
Technically (NDC spec) it can be any JSON value, but the TypeScript SDK actually limits it to an object type (ie Record<string, unknown>) which is honestly more reasonable, so I'm not complaining.

Reply:
Works for me.